### PR TITLE
Added unit tests for Nvidia Card and driver version

### PR DESCRIPTION
**Pull request by KanishkT123**
## Checklist
***This checklist is used to make sure that common guidelines for a pull request are followed.***

- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [X] Title of the pull request is clear and informative.
- [X] If applicable, the code is properly documented.
- [X] The code builds without any errors.
- [X] All unit and integration tests pass.

## Description
Adds a unit test for checking that the environment includes an Nvidia GPU and the correct drivers.

## Related Issues and PRs
Solves #222

### CODE REVIEW

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp [hunk 1]
```diff
@@ -25,6 +33,145 @@ namespace NativeServersUnitTests
...
+
+					hr = pclsObj->Get(L"DriverVersion", 0, &driverNumber, 0, 0);
+					wchar_t *currentDriver = driverNumber.bstrVal;
+					size_t len = wcslen(currentDriver);
+					
+					//Major version number of the card is found at the -7th index
+					std::wstring majorVersion(currentDriver, len - 6, 1);
+					
+					//All drivers from 3.0 onwards support nvencode
+					Assert::IsTrue(std::stoi(majorVersion) > 2);
```
Comment (phongcao) [inline Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Can we use _wtoi to convert wchar_t* to int directly?

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp [hunk 2]
```diff
@@ -25,6 +33,145 @@ namespace NativeServersUnitTests
...
+
+				VariantClear(&vtProp);
+				pclsObj->Release();
+			}
+			//Make sure that we entered the loop
+			Assert::IsTrue(NvidiaPresent);
+			
+			//Clean Up
+			VariantClear(&driverNumber);
+
```
Comment (phongcao) [inline Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: nitpicking: redundant line

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp [hunk 3]
```diff
@@ -25,6 +33,145 @@ namespace NativeServersUnitTests
...
+					NvidiaPresent = true;
+
+					hr = pclsObj->Get(L"DriverVersion", 0, &driverNumber, 0, 0);
+					wchar_t *currentDriver = driverNumber.bstrVal;
+					size_t len = wcslen(currentDriver);
+					
+					//Major version number of the card is found at the -7th index
+					std::wstring majorVersion(currentDriver, len - 6, 1);
+					
+					//All drivers from 3.0 onwards support nvencode
```
Comment (phongcao) [inline Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Can you share me the documents about "All drivers from 3.0 onwards support nvencode"?

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp [hunk 4]
```diff
@@ -25,6 +33,145 @@ namespace NativeServersUnitTests
...
+				{
+					break;
+				}
+
+				VARIANT vtProp;
+				//Finds the manufacturer of the card
+				hr = pclsObj->Get(L"AdapterCompatibility", 0, &vtProp, 0, 0);
+				
+				//Find the nvidia card
+				if (!wcscmp(vtProp.bstrVal, L"NVIDIA")) {
```
Comment (phongcao) [inline Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: nitpicking: { should be put in the next line.

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp [hunk 5]
```diff
@@ -25,6 +33,145 @@ namespace NativeServersUnitTests
...
+					//Set the Nvidia card flag to true
+					NvidiaPresent = true;
+
+					hr = pclsObj->Get(L"DriverVersion", 0, &driverNumber, 0, 0);
+					wchar_t *currentDriver = driverNumber.bstrVal;
+					size_t len = wcslen(currentDriver);
+					
+					//Major version number of the card is found at the -7th index
+					std::wstring majorVersion(currentDriver, len - 6, 1);
```
Comment (phongcao) [inline Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Is it the first or 7th index?

Comment (KanishkT123) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: It's the -7th index (7th character from the end), but it is just one character. The third argument is just the length of the substring we want to make a string.

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp (hunk 3 above)
Comment (KanishkT123) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: This is just extrapolation based on this file: https://github.com/CatalystCode/3dtoolkit/blob/master/Libraries/WebRTC/3dtoolkit_upgrades.patch#L1175

Here the check is to see if the (major version * 16) < 48, i.e. that the major version is 3 or higher.

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp (hunk 5 above)
Comment (phongcao) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Got you, somehow I miss the minus sign.

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp (hunk 1 above)
Comment (KanishkT123) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Done! Much neater :)

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp (hunk 2 above)
Comment (KanishkT123) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Fixed!

File: Samples/Server/NativeServersUnitTests/NativeServersTests.cpp (hunk 4 above)
Comment (KanishkT123) [reply Samples/Server/NativeServersUnitTests/NativeServersTests.cpp]: Fixed!

**Comment by KanishkT123** [comment]
@anderm The error was occurring because I tried to switch from converting a string to int to converting a char to int. C++ doesn't always like that (Unicode issues) and doesn't allow it with the wtoi method. 

So there's no real way to get around having a string conversion, unless we want to load all of boost :O

**Pull request merged**